backend/src/main/java/ca/bc/gov/backendstartapi/endpoint/OrchardEndpoint.java

```
@@ -0,0 +1,75 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+/** This class exposes endpoints for fetching Orchard related information. */
+@RestController
+@RequestMapping(path = "/api/orchards", produces = MediaType.APPLICATION_JSON_VALUE)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Tag(name = "Orchard")
+public class OrchardEndpoint {
+
+  private OrchardService orchardService;
+
+  @Autowired
+  OrchardEndpoint(OrchardService orchardService) {
+    this.orchardService = orchardService;
+  }
+
+  /**
+   * Gets all ParentTree data to an Orchard.
+   *
+   * @param orchardId {@link Orchard}'s identification
+   * @return an {@link OrchardParentTreeDto}
+   * @throws ResponseStatusException if no data is found
+   */
+  @GetMapping(path = "/{orchardId}/parent-tree-genetic-quality")
+  @PreAuthorize("hasRole('user_read')")
+  @Operation(
+      summary = "Fetch the parent tree contribution data to an Orchard.",
+      description = "Returns all parent tree contribution table given an Orchard ID and SPU ID.",
+      responses = {
+        @ApiResponse(responseCode = "200"),
+        @ApiResponse(
+            responseCode = "401",
+            description = "Access token is missing or invalid",
+            content = @Content(schema = @Schema(implementation = Void.class))),
+        @ApiResponse(responseCode = "404", content = @Content(schema = @Schema(hidden = true)))
+      })
+  public OrchardParentTreeDto getParentTreeGeneticQualityData(
+      @PathVariable
+          @Parameter(
+              name = "orchardId",
+              in = ParameterIn.PATH,
+              description = "Identifier of the Orchard.")
+          String orchardId) {
+    Optional<OrchardParentTreeDto> parentTreeDto =
```

If the described behaviour is the default, It would be better to hide this `orElseThrow` within the service itself and throw a custom `ResponseStatusException`  from within it.

If other places will also consume this `findParentTreeGeneticQualityData` and they expect the optional, keep it as is, otherwise, I would recommend encapsulating this error into the service.

---

Good catch! Also this way we're not keeping the code consistent!

---

👍🏻 [Done!](https://github.com/bcgov/nr-spar/pull/249/commits/7c4b956e8cc7cff1a867256557a6cd885da58dfe)